pkg/kudoctl/util/repo/package.go

```diff
@@ -35,25 +35,50 @@ type PackageCRDs struct {
 	Instance         *v1alpha1.Instance
 }
 
-// ReadTarballPackage reads package from tarball and converts it to the CRD format
-func ReadTarballPackage(r io.Reader) (*PackageCRDs, error) {
-	p, err := untarV1Package(r)
+// ReadTarGzPackage reads package from tarball and converts it to the CRD format
+func ReadTarGzPackage(r io.Reader) (*PackageCRDs, error) {
+	p, err := extractV1Package(r)
 	if err != nil {
-		return nil, errors.Wrap(err, "while untarring package")
+		return nil, errors.Wrap(err, "while extracting package files")
 	}
 	return p.getCRDs()
 }
 
 // ReadFileSystemPackage reads package from filesystem and converts it to the CRD format
 func ReadFileSystemPackage(path string) (*PackageCRDs, error) {
-	p, err := fromFilesystem(path)
-	if err != nil {
-		return nil, errors.Wrap(err, "while reading package from filesystem")
+	isTarGz := func() bool {
+		if fi, err := os.Stat(path); err == nil {
+			return fi.Mode().IsRegular() && strings.HasSuffix(path, ".tar.gz")
+		}
+		return false
+	}
+
+	isFolder := func() bool {
+		if fi, err := os.Stat(path); err == nil {
+			return fi.IsDir()
+		}
+		return false
+	}
+
+	switch {
+	case isTarGz():
+		f, err := os.Open(path)
+		if err != nil {
+			return nil, err
+		}
+		return ReadTarGzPackage(f)
```

Comment:
does this method have to be public? It feels like something internal... 🤔

Reply:
It is needed for downloaded packages which come in the form of tar.gz stream. (hence `io.Reader` as a parameter)